### PR TITLE
Add documentation for initializers that build a node from a header and a result builder

### PR DESCRIPTION
**Pull request by ahoppen**
rdar://108643660

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by kimdv — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift [hunk 1]
```diff
@@ -158,6 +226,18 @@ extension SwitchCaseSyntax {
 // So we cannot conform to `HasTrailingCodeBlock` or `HasTrailingMemberDeclBlock`
 
 public extension SwitchExprSyntax {
+  /// Constructs an `switch` expression where `header` builds the text before the opening `{` and `casesBuilder` can be used to build the case items.
```
Comment (bnbarham) [inline Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:229]:
```suggestion
  /// Constructs a `switch` expression where `header` builds the text before the opening `{` and `casesBuilder` can be used to build the case items.
```

File: Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift [hunk 2]
```diff
@@ -158,6 +226,18 @@ extension SwitchCaseSyntax {
...
+  ///
+  /// For example, the following switch expression hase header `switch direction` and two case items for `up` and `down`
```
Comment (bnbarham) [inline Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:231]:
```suggestion
  /// For example, the following switch expression has header `switch direction` and two case items for `up` and `down`
```

File: Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift [hunk 3]
```diff
@@ -132,6 +173,30 @@ public extension IfExprSyntax {
...
+  /// if x == 1 {
+  ///     return "one
+  /// } else if x == 2 {
+  ///     return "two
+  /// } else {
+  ///     return "many
+  /// }
+  /// ```
+  ///
+  /// You can use
```
Comment (bnbarham) [inline Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:189]: I like how this one has an actual example of how to construct it. How would you feel about adding something similar to the others? If not that's fine, some documentation is better than none :). But a lot of the `For example the following ... has ...` could just be code instead.

File: Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift [hunk 4]
```diff
@@ -95,6 +113,16 @@ extension InitializerDeclSyntax: HasTrailingOptionalCodeBlock {}
 public protocol HasTrailingMemberDeclBlock {
   var memberBlock: MemberDeclBlockSyntax { get set }
 
+  /// Constructs a syntax node where `header` builds the text of the node before the members in braces and `membersBuilder` is used to list the node’s members.
+  ///
+  /// For example the following function has header `struct Point` and the body will contain two `DeclSyntax` with text `var x: Int` and `var y: Int`, respecitively.
```
Comment (bnbarham) [inline Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:118]:
```suggestion
  /// For example the following function has header `struct Point` and the body will contain two `DeclSyntax` with text `var x: Int` and `var y: Int`, respectively.
```

File: Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift (hunk 3 above)
Comment (ahoppen) [reply Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:189]: Added more examples

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Pull request merged**